src/tools/annotation/CircleRoiTool.js

```diff
@@ -461,22 +494,51 @@ function _calculateStats(image, element, handles, modality, pixelSpacing) {
 
   if (modality === 'PT') {
     meanStdDevSUV = {
-      mean: calculateSUV(image, ellipseMeanStdDev.mean, true) || 0,
-      stdDev: calculateSUV(image, ellipseMeanStdDev.stdDev, true) || 0,
+      mean: measurementUncertainty.getGenericRounding(
+        calculateSUV(image, ellipseMeanStdDev.mean, true)
+      ),
+      stdDev: measurementUncertainty.getGenericRounding(
+        calculateSUV(image, ellipseMeanStdDev.stdDev, true)
+      ),
     };
   }
 
+  const radius =
+    (circleCoordinates.width *
+      ((pixelSpacing && pixelSpacing.colPixelSpacing) || 1)) /
+    2;
+
+  const perimeter = new Decimal(2 * Math.PI * radius) || 0;
+
+  const diameter = radius * 2 || 0;
+
+  const pixelDiagonal = measurementUncertainty.getPixelDiagonal(
+    pixelSpacing.colPixelSpacing,
+    pixelSpacing.rowPixelSpacing
+  );
+
+  const areaUncertainty = perimeter * pixelDiagonal || 0;
+
+  const diameterUncertainty = new Decimal(pixelDiagonal * Decimal.sqrt(2)) || 0;
+
   const area =
     Math.PI *
     ((circleCoordinates.width * (pixelSpacing.colPixelSpacing || 1)) / 2) *
     ((circleCoordinates.height * (pixelSpacing.rowPixelSpacing || 1)) / 2);
 
   return {
-    area: area || 0,
+    area: measurementUncertainty.roundArea(area, areaUncertainty),
+    areaUncertainty: measurementUncertainty.roundUncertainty(areaUncertainty),
+    radius: radius || 0,
+    diameter: measurementUncertainty.roundArea(diameter, diameterUncertainty),
+    diameterUncertainty: measurementUncertainty.roundUncertainty(
+      diameterUncertainty
+    ),
+    perimeter,
     count: ellipseMeanStdDev.count || 0,
-    mean: ellipseMeanStdDev.mean || 0,
+    mean: measurementUncertainty.getGenericRounding(ellipseMeanStdDev.mean),
```

Comment:
I think we would still need to use the `|| 0` fallback because your method `getGenericRounding` could return an `undefined`